content/docs/rfcs/72/README.md

```diff
@@ -0,0 +1,297 @@
+uses [32/RLN-V1](/spec/32/) as a spam-prevention mechanism by generating zero-knowledge proofs.
+
+# Background
+The secure storage of keys is important in peer-to-peer messaging applications.
+A `72/WAKU-RLN-KEYSTORE` uses zero-knowledge proofs for anonymous rate-limiting for messaging frameworks. 
+Generated credentials by a user are encrypted and stored in the keystore to be retrieved over a network.
+With [32/RLN-V1](/spec/32/), sending and receiving 
+messages will ensure a message rate for a network is being followed while preserving the anonymity of the message owner. 
+
+## Waku RLN Keystore Format:
+
+A format example of a keystore used by a [17/WAKU2-RLN-Relay](/spec/17/).
+
+```js
+const Keystore {
+	application: "waku-rln-relay" ,
+	appIdentifier: "string",
+	version: "string",
+	credentials: {
+    		"membershipHash": {
+      			crypto: {
+        			cipher: "string",
+        			cipherparams: {
+					iv: "string",
+				},
+        			ciphertext: "string",
+        			kdf: "string",
+        			kdfparams: {
+          				dklen: integer,
+          				c: integer,
+          				prf: "string",
+          				salt: "string",
+        			},
+        			mac: "string",
+      			}
+    		}
+	}
+}
+
+```
+
+# Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, 
+“NOT RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+The keystore MUST be generated by a cryptographic construction with password verification and decryption.
+
+Keystore modules MUST include metadata, key derivation function, checksum, cipher, and a membership hash.
+
+## Metadata:
+Information about the keystore SHOULD be stored in the metadata.
+
+The declaration of `application`, `version`, and `appIdentifier` COULD occur in the metadata.
+
+- `application` : current application, MUST be a string
+
+- `version` : application version, MUST be a string
+
+- `appIdentifier`: application identifier, MUST be a string
+
+## Credentials:
+
+After RLN credentials are generated, it MUST be stored in a JSON schema.
+The Waku RLN credentials MUST consist of a `membershipHash` and `WakuCredential`.
+The `membershipHash` will be an identity hash of the user.
+The `WakuCredential` will store to encryption portion of the keystore.
+There COULD be multiple credentials stored in a keystore, categorized by the `membershipHash`.
+
+Each contruct MUST include the keypair:
+> key: [`membershipHash`]: pair: [`WakuCredential`]
+
+### membershipHash 
+
+The `membershipHash` SHOULD be generated by user's participating in a membership group.
+Each user SHOULD register to the group with an `identity_commitment` stored in a Merkle tree.
+A cryptographic hash function that SHOULD be used to generate the `membershipHash` is [SHA256](https://www.rfc-editor.org/rfc/rfc4634.txt), 
+other hash functions MAY be used.
```

Comment:
if other hash functions are used, it should be indicated via the version string or other parameters

Reply:
@rymnc which version sting are you refering to, the `version` in the metadata?

Reply:
correct, actually i noticed later in the rfc we define a field "hashFunction" which takes care of this. feel free to ignore this request for change!